TP3 & master: JavaScript Error when opening pages.

This happens on fresh install of the current TP3 image (i.e. on that 1.2.6):

![image](https://user-images.githubusercontent.com/99080/35480578-6e194e8a-0411-11e8-9c7a-a898497dfe50.png)

Not sure whether this is an issue on the build or in general. @dsimansk do you see something like this on the productised images, too ?

BTW, this is how the UI image is built (based on the tag `1.2.5` or `1.2.6`):

<details><summary>Build Log</summary>

```
++ ./mvnw -Dmaven.repo.local=/tmp/clean-repo --batch-mode -Prelease,image,flash -Dfabric8.mode=kubernetes -f
 ui package
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building UI 1.2.5
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- frontend-maven-plugin:1.6:install-node-and-npm (install node and npm) @ syndesis-ui ---
[INFO] Node v8.9.1 is already installed.
[INFO]
[INFO] --- frontend-maven-plugin:1.6:install-node-and-yarn (install yarn) @ syndesis-ui ---
[INFO] Node v8.9.1 is already installed.
[INFO] Yarn 1.2.1 is already installed.
[INFO]
[INFO] --- frontend-maven-plugin:1.6:yarn (yarn install) @ syndesis-ui ---
[INFO] Running 'yarn install --no-progress' in /Users/roland/Development/syndesis/syndesis/app/ui
[INFO] yarn install v1.2.1
[INFO] [1/5] Resolving packages...
[INFO] success Already up-to-date.
[INFO] Done in 0.76s.
[INFO]
[INFO] --- frontend-maven-plugin:1.6:yarn (yarn cache dir) @ syndesis-ui ---
[INFO] Running 'yarn cache dir' in /Users/roland/Development/syndesis/syndesis/app/ui
[INFO] /Users/roland/Library/Caches/Yarn/v1
[INFO]
[INFO] --- frontend-maven-plugin:1.6:yarn (yarn ng build) @ syndesis-ui ---
[INFO] Running 'yarn build:ci' in /Users/roland/Development/syndesis/syndesis/app/ui
[INFO] yarn run v1.2.1
[INFO] $ yarn ng build --aot --prod --progress=false
[ERROR] Template parse warnings:
[ERROR] The <template> element is deprecated. Use <ng-template> instead ("
[ERROR] [WARNING ->]<template [ngIf]="!isClosed">
[ERROR]   <div [class]="'alert alert-' + type" role="alert" [ngClass]="classes""): ng:///Users/roland/Development/syndesis/syndesis/app/ui/node_modules/ngx-bootstrap/alert/alert.component.d.ts.AlertComponent.html@1:0
[INFO] Hash: c55a9e8da2242a8d2be0
[INFO] Time: 90363ms
[INFO] chunk    {0} 0.040409d0228411e430ad.chunk.js (common) 353 kB {1} {2} {3} {4} {5} {6} {7} {8} {9} {10} [rendered]
[INFO] chunk    {1} 1.51fb4e3b5773d3e33757.chunk.js 936 kB {2} {3} {4} {5} {6} {8} [rendered]
[INFO] chunk    {2} 2.9b3cfecc63d0c7e0cd95.chunk.js 458 kB {1} {3} {4} {5} {6} {8} [rendered]
[INFO] chunk    {3} 3.3311e2b168984581ec8a.chunk.js 294 kB {1} {2} {4} {5} {6} {8} [rendered]
[INFO] chunk    {4} 4.4d82602ff625cbf5ebfa.chunk.js 225 kB {1} {2} {3} {5} {6} {8} [rendered]
[INFO] chunk    {5} 5.9b871f1ecb23ebafd239.chunk.js 56 kB {1} {2} {3} {4} {6} {8} [rendered]
[INFO] chunk    {6} 6.eff94fe20e6ff2e9e9b8.chunk.js 20.9 kB {1} {2} {3} {4} {5} {8} [rendered]
[INFO] chunk    {7} polyfills.d83e4e39ab15798252e2.bundle.js (polyfills) 240 kB {12} [initial] [rendered]
[INFO] chunk    {8} main.8721ccc1fce605e9161a.bundle.js (main) 183 kB {11} [initial] [rendered]
[INFO] chunk    {9} scripts.0128b0b14eb92aafd3ae.bundle.js (scripts) 962 kB {12} [initial] [rendered]
[INFO] chunk   {10} styles.a8c6d0afd78c21d11711.bundle.css (styles) 228 bytes {12} [initial] [rendered]
[INFO] chunk   {11} vendor.ac4e6acdacc1a250d00c.bundle.js (vendor) 7.39 MB [initial] [rendered]
[INFO] chunk   {12} inline.21b5288893cb454433bf.bundle.js (inline) 0 bytes [entry] [rendered]
[INFO] Done in 97.96s.
[INFO]
[INFO] --- maven-assembly-plugin:3.0.0:single (dist) @ syndesis-ui ---
[INFO] Reading assembly descriptor: src/assembly/unix-dist.xml
[WARNING] The assembly descriptor contains a filesystem-root relative reference, which is not cross platform compatible /srv
[WARNING] The assembly descriptor contains a filesystem-root relative reference, which is not cross platform compatible /etc/
[INFO] Building tar: /Users/roland/Development/syndesis/syndesis/app/ui/target/syndesis-ui-1.2.5-dist.tar.gz
[WARNING] The assembly descriptor contains a filesystem-root relative reference, which is not cross platform compatible /srv
[WARNING] The assembly descriptor contains a filesystem-root relative reference, which is not cross platform compatible /etc/
[INFO] Building zip: /Users/roland/Development/syndesis/syndesis/app/ui/target/syndesis-ui-1.2.5-dist.zip
[INFO]
[INFO] --- fabric8-maven-plugin:3.5.33:build (exec) @ syndesis-ui ---
[INFO] F8: Building Docker image in Kubernetes mode
[WARNING] F8: Cannot access cluster for detecting mode: Network is unreachable (connect failed)
[INFO] F8: Generators:
[INFO] F8:  - spring-boot
[INFO] F8:  - wildfly-swarm
[INFO] F8:  - karaf
[INFO] F8:  - vertx
[INFO] F8:  - java-exec
[INFO] F8:  - webapp
[WARNING] F8: Cannot access cluster for detecting mode: Network is unreachable (connect failed)
[INFO] Copying files to /Users/roland/Development/syndesis/syndesis/app/ui/target/docker/syndesis/syndesis-ui/1.2.5/build/maven
[WARNING] F8: Dockerfile /Users/roland/Development/syndesis/syndesis/app/ui/docker/Dockerfile does not contain an ADD or COPY directive to include assembly created at maven. Ignoring assembly.
[INFO] Building tar: /Users/roland/Development/syndesis/syndesis/app/ui/target/docker/syndesis/syndesis-ui/1.2.5/tmp/docker-build.tar
[INFO] F8: [syndesis/syndesis-ui:1.2.5]: Created docker-build.tar in 288 milliseconds
[INFO] F8: Step 1/15 : FROM docker.io/centos:7
[INFO] F8:
[INFO] F8: ---> 3fa822599e10
[INFO] F8: Step 2/15 : ENV NGINX_VERSION 1.13.4-1.el7
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> b23ff7ebc8ae
[INFO] F8: Step 3/15 : ARG SRC_DIR="maven/dist"
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 2d67c7a20773
[INFO] F8: Step 4/15 : ARG CONTEXT_DIR="."
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 41d0065f4a27
[INFO] F8: Step 5/15 : LABEL name="nginxinc/nginx"       maintainer="Syndesis Authors <syndesis@googlegroups.com>"       vendor="NGINX Inc."       version="${NGINX_VERSION}"       release="1"       summary="NGINX"       description="nginx will do ....."
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 94c272b49f5b
[INFO] F8: Step 6/15 : LABEL url="https://www.nginx.com/"       io.k8s.display-name="NGINX"       io.openshift.expose-services="http:8080"       io.openshift.tags="nginx,nginxinc"
[INFO] F8:
[INFO] F8: ---> Using cache
.com>"       vendor="NGINX Inc."       version="${NGINX_VERSION}"       release="1"       summary="NGINX"
    description="nginx will do ....."
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 94c272b49f5b
[INFO] F8: Step 6/15 : LABEL url="https://www.nginx.com/"       io.k8s.display-name="NGINX"       io.openshi
ft.expose-services="http:8080"       io.openshift.tags="nginx,nginxinc"
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> f9da7eafe756
[INFO] F8: Step 7/15 : ADD ${CONTEXT_DIR}/nginx.repo /etc/yum.repos.d/nginx.repo
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 89dbee96163d
[INFO] F8: Step 8/15 : RUN curl -sO http://nginx.org/keys/nginx_signing.key &&     rpm --import ./nginx_signing.key &&     yum -y install --setopt=tsflags=nodocs nginx-${NGINX_VERSION}.ngx &&     rm -f ./nginx_signing.key &&     yum clean all
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> f509cca26a00
[INFO] F8: Step 9/15 : RUN ln -sf /dev/stdout /var/log/nginx/access.log &&     ln -sf /dev/stderr /var/log/nginx/error.log &&     sed -i 's/\/var\/run\/nginx.pid/\/var\/cache\/nginx\/nginx.pid/g' /etc/nginx/nginx.conf &&     sed -i -e '/user/!b' -e '/nginx/!b' -e '/nginx/d' /etc/nginx/nginx.conf &&     rm -f /etc/nginx/conf.d/default.conf &&     chown -R 998 /var/cache/nginx /etc/nginx &&     chmod -R g=u /var/cache/nginx /etc/nginx
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 1def5f54f71d
[INFO] F8: Step 10/15 : EXPOSE 8080 8443
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 9dca0a587986
[INFO] F8: Step 11/15 : RUN ln -sf /usr/share/nginx/html/config/config.json /usr/share/nginx/html/config.json
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 3262942a6919
[INFO] F8: Step 12/15 : USER 998
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 88f0ba4ff239
[INFO] F8: Step 13/15 : CMD ["nginx", "-g", "daemon off;"]
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> 0a3dd2977e01
[INFO] F8: Step 14/15 : COPY ${CONTEXT_DIR}/nginx-syndesis.conf /etc/nginx/conf.d
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> d4781f696432
[INFO] F8: Step 15/15 : COPY ${SRC_DIR} /usr/share/nginx/html
[INFO] F8:
[INFO] F8: ---> Using cache
[INFO] F8: ---> b6254e1907f2
[INFO] F8: Successfully built b6254e1907f2
[INFO] F8: Successfully tagged syndesis/syndesis-ui:1.2.5
[INFO] F8: [syndesis/syndesis-ui:1.2.5]: Built image sha256:b6254
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

</details>

--- a/app/ui/src/app/app.component.ts
+++ b/app/ui/src/app/app.component.ts
@@ -168,14 +168,14 @@ export class AppComponent implements OnInit, AfterViewInit {
   }
 
   startTour() {
-    this.tourService.start();
+    //this.tourService.start();
     this.userService.setTourState(true);
     this.guidedTourStatus = true;
   }
 
   endTour() {
     if (!!this.guidedTourStatus) {
-      this.tourService.end();
+      //this.tourService.end();
       this.userService.setTourState(false);
       this.guidedTourStatus = false;
     }

--- a/app/ui/src/app/connections/create-page/connection-basics/connection-basics.component.ts
+++ b/app/ui/src/app/connections/create-page/connection-basics/connection-basics.component.ts
@@ -34,6 +34,7 @@ export class ConnectionsConnectionBasicsComponent implements OnInit {
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([{
           anchorId: 'connections.type',
@@ -45,6 +46,7 @@ export class ConnectionsConnectionBasicsComponent implements OnInit {
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 
   onSelected(connector: Connector) {

--- a/app/ui/src/app/connections/create-page/review/review.component.ts
+++ b/app/ui/src/app/connections/create-page/review/review.component.ts
@@ -102,6 +102,7 @@ export class ConnectionsReviewComponent implements CanComponentDeactivate, OnIni
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([ {
         route: 'connections/create/review',
@@ -113,6 +114,7 @@ export class ConnectionsReviewComponent implements CanComponentDeactivate, OnIni
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 
   ngOnDestroy() {

--- a/app/ui/src/app/dashboard/dashboard.component.ts
+++ b/app/ui/src/app/dashboard/dashboard.component.ts
@@ -39,6 +39,7 @@ export class DashboardComponent implements OnInit {
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([ {
         route: 'dashboard',
@@ -58,5 +59,6 @@ export class DashboardComponent implements OnInit {
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 }

--- a/app/ui/src/app/dashboard/emptystate.component.ts
+++ b/app/ui/src/app/dashboard/emptystate.component.ts
@@ -41,6 +41,7 @@ export class EmptyStateComponent implements OnInit {
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([
         {
@@ -53,5 +54,6 @@ export class EmptyStateComponent implements OnInit {
       ]);
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 }

--- a/app/ui/src/app/integrations/edit-page/action-configure/action-configure.component.ts
+++ b/app/ui/src/app/integrations/edit-page/action-configure/action-configure.component.ts
@@ -289,6 +289,7 @@ export class IntegrationsConfigureActionComponent extends FlowPage
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([ {
           anchorId: 'integrations.done',
@@ -300,6 +301,7 @@ export class IntegrationsConfigureActionComponent extends FlowPage
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 
   ngOnDestroy() {

--- a/app/ui/src/app/integrations/edit-page/action-select/action-select.component.ts
+++ b/app/ui/src/app/integrations/edit-page/action-select/action-select.component.ts
@@ -132,6 +132,7 @@ export class IntegrationsSelectActionComponent extends FlowPage
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([ {
         anchorId: 'actions.available',
@@ -142,6 +143,7 @@ export class IntegrationsSelectActionComponent extends FlowPage
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 
   ngOnDestroy() {

--- a/app/ui/src/app/integrations/edit-page/connection-select/connection-select.component.ts
+++ b/app/ui/src/app/integrations/edit-page/connection-select/connection-select.component.ts
@@ -122,6 +122,7 @@ export class IntegrationsSelectConnectionComponent extends FlowPage implements O
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([{
         anchorId: 'integrations.panel',
@@ -139,6 +140,7 @@ export class IntegrationsSelectConnectionComponent extends FlowPage implements O
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 
   ngOnDestroy() {

--- a/app/ui/src/app/integrations/edit-page/flow-view/flow-view.component.ts
+++ b/app/ui/src/app/integrations/edit-page/flow-view/flow-view.component.ts
@@ -237,6 +237,7 @@ export class FlowViewComponent extends ChildAwarePage
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([{
         anchorId: 'integrations.step',
@@ -247,6 +248,7 @@ export class FlowViewComponent extends ChildAwarePage
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 
   ngOnDestroy() {

--- a/app/ui/src/app/integrations/edit-page/integration-basics/integration-basics.component.ts
+++ b/app/ui/src/app/integrations/edit-page/integration-basics/integration-basics.component.ts
@@ -78,6 +78,7 @@ export class IntegrationBasicsComponent extends FlowPage implements OnInit {
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([{
           anchorId: 'integrations.publish',
@@ -90,5 +91,6 @@ export class IntegrationBasicsComponent extends FlowPage implements OnInit {
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 }

--- a/app/ui/src/app/integrations/edit-page/save-or-add-step/save-or-add-step.component.ts
+++ b/app/ui/src/app/integrations/edit-page/save-or-add-step/save-or-add-step.component.ts
@@ -160,6 +160,7 @@ export class IntegrationsSaveOrAddStepComponent extends FlowPage implements OnIn
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize([ {
         anchorId: 'integrations.step',
@@ -171,5 +172,6 @@ export class IntegrationsSaveOrAddStepComponent extends FlowPage implements OnIn
       );
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 }

--- a/app/ui/src/app/integrations/list/list.component.html
+++ b/app/ui/src/app/integrations/list/list.component.html
@@ -52,7 +52,7 @@
                  let-integration="item"
                  let-index="index">
       <pfng-action class="list-pf-actions"
-                   [config]="getActionConfig(integration)"
+                   [config]="actionConfigs[integration.id]"
                    (onActionSelect)="requestAction($event.id, integration)">
       </pfng-action>
     </ng-template>

--- a/app/ui/src/app/integrations/list/list.component.ts
+++ b/app/ui/src/app/integrations/list/list.component.ts
@@ -1,4 +1,4 @@
-import { ApplicationRef, Component, Input, ViewChild } from '@angular/core';
+import { ApplicationRef, Component, Input, OnChanges, OnInit, ViewChild } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import { ModalDirective } from 'ngx-bootstrap/modal';
 import { Subscription } from 'rxjs/Subscription';
@@ -19,15 +19,20 @@ import { IntegrationViewBase } from '../components';
 import { ModalService, NotificationService } from '@syndesis/ui/common';
 import { log, getCategory } from '@syndesis/ui/logging';
 
+interface ActionConfigs {
+  [id: string]: ActionConfig;
+}
+
 @Component({
   selector: 'syndesis-integrations-list',
   templateUrl: './list.component.html',
   styleUrls: ['./list.component.scss']
 })
-export class IntegrationsListComponent extends IntegrationViewBase {
+export class IntegrationsListComponent extends IntegrationViewBase implements OnInit, OnChanges {
   @Input() complete: boolean;
   @Input() integrations: Integrations = [];
   listConfig: ListConfig;
+  actionConfigs: ActionConfigs;
 
   constructor(
     public store: IntegrationStore,
@@ -39,6 +44,27 @@ export class IntegrationsListComponent extends IntegrationViewBase {
     integrationSupportService: IntegrationSupportService,
   ) {
     super(store, route, router, notificationService, modalService, application, integrationSupportService);
+  }
+
+  handleAction($event: Action, item: any) {
+    if ($event.id === 'createIntegration') {
+      this.router.navigate(['/integrations/create']);
+    }
+  }
+
+  handleClick($event: ListEvent) {
+    this.router.navigate(['/integrations', $event.item.id], {
+      relativeTo: this.route
+    });
+  }
+
+  ngOnChanges(changes: any) {
+    if (changes.integrations) {
+      this.actionConfigs = this.generateActionConfigMap(changes.integrations.currentValue);
+    }
+  }
+
+  ngOnInit() {
     this.listConfig = {
       dblClick: false,
       multiSelect: false,
@@ -61,21 +87,19 @@ export class IntegrationsListComponent extends IntegrationViewBase {
         } as ActionConfig
       } as EmptyStateConfig
     };
+    this.actionConfigs = this.generateActionConfigMap(this.integrations);
   }
 
-  handleAction($event: Action, item: any) {
-    if ($event.id === 'createIntegration') {
-      this.router.navigate(['/integrations/create']);
+  private generateActionConfigMap(integrations: Integrations) {
+    const answer: ActionConfigs = {};
+    for (const integration of integrations) {
+      const actionConfig = this.getActionConfig(integration);
+      answer[integration.id] = actionConfig;
     }
+    return answer;
   }
 
-  handleClick($event: ListEvent) {
-    this.router.navigate(['/integrations', $event.item.id], {
-      relativeTo: this.route
-    });
-  }
-
-  getActionConfig(integration: Integration): ActionConfig {
+  private getActionConfig(integration: Integration): ActionConfig {
     const actionConfig = {
       primaryActions: [],
       moreActions: [

--- a/app/ui/src/app/settings/oauth-apps/oauth-apps.component.ts
+++ b/app/ui/src/app/settings/oauth-apps/oauth-apps.component.ts
@@ -125,6 +125,7 @@ export class OAuthAppsComponent implements OnInit {
     /**
      * If guided tour state is set to be shown (i.e. true), then show it for this page, otherwise don't.
      */
+    /*
     if (this.userService.getTourState() === true) {
       this.tourService.initialize(
         [
@@ -145,5 +146,6 @@ export class OAuthAppsComponent implements OnInit {
 
       setTimeout(() => this.tourService.start());
     }
+    */
   }
 }

